### ISSUE TITLE
[Feature Request] Export automatically after making a change in my account 

### ISSUE THREAD

**Issue by Terkyz** [nlogozzo/NickvisionMoney#334]
Basically, I would appreciate if possible the ability to automatically export a copy of my account after making a change (transaction) to a previously defined directory. 

I mainly request this to have a backup on my Nextcloud server, but it could also serve if I wanted to synchronize Denaro with any mobile apps that support CSV.

**Comment by nlogozzo** [comment]
So you'd just like to it automatically export a CSV whenever something changes in the account to a predefined directory, correct? Seems like something we could do...

**Comment by Terkyz** [comment]
That's right. I just didn't know how to describe it as well as you did. 🙃

**Comment by fsobolev** [comment]
@nlogozzo We need to think how this feature should work and how it should be implemented in UI considering what I proposed in #245

**Comment by nlogozzo** [comment]
I was thinking we could have a simple preferences option, `off` by default` that says `Backup Account to CSV`. `If checked, this option will export the entire account to a csv file when a change is made.`
And if `Backup Account to CSV` is on, there will be a row right under it that will ask for the directory to save to.

**Comment by fsobolev** [comment]
> I was thinking we could have a simple preferences option, `off` by default that says `Backup Account to CSV`. `If checked, this option will export the entire account to a csv file when a change is made.` And if `Backup Account to CSV` is on, there will be a row right under it that will ask for the directory to save to.

I'm not sure about additional rows, maybe 1 row with buttons similar to what we have with receipts will be good, we'll see. But overall, okay, that will work without affecting #245.

**Comment by nlogozzo** [comment]
Here's how the setting will work: 
![image](https://user-images.githubusercontent.com/17648453/225653896-9855aa4a-432d-4041-93a9-58f6f87a0648.png)

Now it has to be an application-wide setting instead of a per-account setting because if a backup folder is set on an account and that account is opened on another computer, that folder obviously won't be there and that setting will make no sense. So it'd have to be an application-wide setting, which is a per-computer basis.

**Comment by nlogozzo** [comment]
@Terkyz Feel free to manually install [this flatpak file](https://github.com/nlogozzo/NickvisionMoney/actions/runs/4441304821) and test it to make sure it meets all your needs for this feature :)

To install:
- Run `flatpak install org.nickvision.money.flatpak` from the directory containing the .flatpak file
- Run `flatpak run org.nickvision.money//master` to start the app!

**Comment by nlogozzo** [comment]
https://github.com/nlogozzo/NickvisionMoney/releases/tag/2023.4.0-beta1
